### PR TITLE
feat(auth): add auth service interface with check method

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

No way to check auth tokens.
Fixes: N/A

## What is the new behavior?
An interface is present for checking auth tokens.

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/auth/src/drivers/interfaces/auth-service.interface.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+export interface DaffAuthServiceInterface<
+  TRequest extends DaffAuthToken,
+> {
+  /**
+   * Checks the validity of an auth token.
+   *
+   * @returns Whether the auth token is valid.
+   */
+  check(auth: TRequest): Observable<boolean>;
```
Comment (damienwebdev) [inline libs/auth/src/drivers/interfaces/auth-service.interface.ts]: This should not be generic. `check` should probably take no arguments and return void. 

We'll be adding the token to the relevant requests as a header, and we can assume a successful response is true and an error response (403) is false.

File: libs/auth/testing/src/drivers/in-memory/auth/auth.service.spec.ts [hunk 2]
```diff
@@ -41,16 +42,20 @@ describe('Driver | InMemory | Auth | AuthService', () => {
       httpMock.verify();
     });
 
-    it('should send a post request and return true', () => {
-      service.check(mockAuth).subscribe(res => {
-        expect(res).toEqual(true);
+    it('should send a post request and not throw an error', () => {
+      service.check().pipe(
+        catchError((err, caught) => {
```
[the remark this answers is not in the document]
Comment (griest024) [reply libs/auth/testing/src/drivers/in-memory/auth/auth.service.spec.ts]: Nothing actually.

[the remark this answers is not in the document]
Comment (griest024) [reply libs/auth/testing/src/drivers/in-memory/auth/auth.service.spec.ts]: I have removed the `catchError` and added a `done` callback so the test will wait for the expectation.

[the remark this answers is not in the document]
Comment (damienwebdev) [reply libs/auth/testing/src/drivers/in-memory/auth/auth.service.spec.ts]: This seems to have snuck back in?

[the remark this answers is not in the document]
Comment (griest024) [reply libs/auth/testing/src/drivers/in-memory/auth/auth.service.spec.ts]: No I changed the testing driver test and mistakenly didn't change this one. Pushing this change now.

**Comment by griest024** [comment]
@damienwebdev newsletter tests are failing the build:
```
ERROR: 'Spec 'DaffNewsletterHubspotService when sending should return an observable of HubspotResponse' has no expectations.'
HeadlessChrome 80.0.3987 (Linux 0.0.0): Executed 22 of 38 SUCCESS (0 secs / 0.29 secs)
HeadlessChrome 80.0.3987 (Linux 0.0.0) ERROR
  An error was thrown in afterAll
  Uncaught TypeError: actual.subscribe is not a function thrown
```

**Pull request merged**